src/system/logger.rs

```diff
@@ -1,150 +1,102 @@
 use config::Config;
-use log;
-use log::{info, Level};
-
+use slog::{info, Drain, Logger, Never, OwnedKVList, Record, Level, o};
+use std::str::FromStr;
 use crate::actor::{
     Actor, ActorRef, All, BasicActorRef, BoxActorProd, ChannelMsg, Context, DeadLetter, Props,
     Subscribe, Tell,
 };
 
-pub type LogActor = Box<dyn Actor<Msg = LogEntry> + Send>;
 
 #[derive(Clone)]
-pub struct Logger {
-    level: Level,
-    actor: ActorRef<LogEntry>,
+pub struct LoggerConfig {
+    time_fmt: String,
+    date_fmt: String,
+    log_fmt: String,
+    filter: Vec<String>,
+    level: Level
 }
 
-impl Logger {
-    pub fn init(level: Level, actor: ActorRef<LogEntry>) -> Self {
-        let logger = Logger { level, actor };
-
-        let _ = log::set_boxed_logger(Box::new(logger.clone())); // result is Err for some reason (.unwrap() panics)
-        log::set_max_level(level.to_level_filter());
-
-        logger
+impl<'a> From<&'a Config> for LoggerConfig {
+    fn from(config: &Config) -> Self {
+        LoggerConfig {
+            time_fmt: config.get_str("log.time_format").unwrap().to_string(),
+            date_fmt: config.get_str("log.date_format").unwrap().to_string(),
+            log_fmt: config.get_str("log.log_format").unwrap().to_string(),
+            filter: config
+                .get_array("log.filter")
+                .unwrap_or(vec![])
+                .into_iter()
+                .map(|e| e.to_string())
+                .collect(),
+            level: config
+                .get_str("log.level")
+                .map(|l| Level::from_str(&l).unwrap_or(Level::Info))
+                .unwrap_or(Level::Info)
+        }
     }
 }
 
-impl log::Log for Logger {
-    fn enabled(&self, metadata: &log::Metadata) -> bool {
-        metadata.level() <= self.level
-    }
+pub(crate) fn default_log(cfg: &Config) -> Logger {
+    let cfg = LoggerConfig::from(cfg);
 
-    fn log(&self, record: &log::Record) {
-        self.actor.tell(LogEntry::from(record), None);
-    }
+    let drain = DefaultConsoleLogger::new(cfg.clone()).filter_level(cfg.level).fuse();
+    let logger = Logger::root(drain, o!());
 
-    fn flush(&self) {}
-}
+    let _scope_guard = slog_scope::set_global_logger(logger.clone());
```

Comment:
Don't these guards need to remain undropped as long as the logger's being used?

Reply:
Good catch. I will create a new PR to address this.